app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -101,22 +133,26 @@ def parse_az(az)
       end
 
       def parse_instance(instance)
-        uid         = "#{@subscription_id}\\#{instance.fetch_path('resourceGroup')}\\#{instance.fetch_path('name')}"
+        uid = resource_uid(@subscription_id,
+                           instance.fetch_path('resourceGroup'),
+                           instance.fetch_path('type').downcase,
+                           instance.fetch_path('name'))
```

Comment:
I would love to use a GUID to compose the `uid`, but we cannot find a common GUID for the same instance during instance parsing and stack resource parsing. Good thing is the `uid` used during refresh is transient. We should have no problem for name collision.